src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/ChatWithCustomData/ChatWithCustomData.Web/README.md

```diff
@@ -0,0 +1,209 @@
+# AI Chat with Custom Data
+
+This project is an AI chat application that demonstrates how to chat with custom data using an AI language model. This was created using an early preview version of a new AI Starter Web App template for .NET, thanks for trying it out!
+
+>[!NOTE]
+> Before you can run this project you'll need to setup your environment to work with the providers you configured in the template options. See below for the details specific to your choices.
+
+#### ---#if (UseAzure)
+### Prerequisites
+To use Azure OpenAI or Azure AI Search, you will need an Azure account. If you don't have one already, [create an Azure account](https://azure.microsoft.com/free/).
+
+#### ---#endif
+# Configure the AI Model Provider
+
+#### ---#if (IsOpenAI)
+## Using OpenAI
+
+To call the OpenAI REST API, you will need an API key. To obtain one, first [create a new OpenAI account](https://platform.openai.com/signup) or [log in](https://platform.openai.com/login). Next, navigate to the API key page and select "Create new secret key", optionally naming the key. Make sure to save your API key somewhere safe and do not share it with anyone.
+
+#### ---#if (hostIdentifier == "vs")
+Configure your API key for this project, using .NET User Secrets:
+
+1. In Visual Studio, right-click on your project in the Solution Explorer and select "Manage User Secrets".
+2. This will open a secrets.json file where you can store your API key securely, add the following key & value to the file:
+   
+```json
+{
+   "OpenAI:Key": "YOUR-API-KEY"
+}
+```
+
+#### ---#else
+From the command line, configure your API key for this project using .NET User Secrets by running the following commands:
+
+```sh
+cd <<your-project-directory>>
+dotnet user-secrets set OpenAI:Key YOUR-API-KEY
+```
+
+#### ---#endif
+#### ---#endif
+#### ---#if (IsOllama)
+## Setting up a local environment using Ollama
+To enable a fully offline/local workflow, the .NET AI templates support Ollama, an application that allows you to run a set of AI models locally on your machine. Note: Ollama is an excellent open source product, but it is not maintained by Microsoft.
+
+### 1. Install Ollama & Run
+First, download and install Ollama from their [official website](https://www.ollama.com). Follow the installation instructions specific to your operating system.
+
+### 2. Choose and Install Models
+This project uses the `llama3.2` and `all-minilm` language modes. To install these models, use the following commands in your terminal once Ollama has been started:
```

Comment:
```suggestion
This project uses the `llama3.2` and `all-minilm` language modes. To install these models, use the following commands in your terminal once Ollama has been installed:
```

(Since for these commands it doesn't matter whether Ollama is already running)